staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -282,10 +284,9 @@ func ParseWatchCacheSizes(cacheSizes []string) (map[schema.GroupResource]int, er
 		if err != nil {
 			return nil, fmt.Errorf("invalid size of watch cache size: %s", c)
 		}
-		if size < 0 {
-			return nil, fmt.Errorf("watch cache size cannot be negative: %s", c)
+		if size < -1 {
```

Comment:
I think we need to introduce some consistency.
From the above code it seems that for 0 we also just use undecorated storage. So do we really need -1 too?
Maybe what we should do is:
- 1<= - watchcache with a given size
- 0 - underocated
- -1>= - forbidden?

I don't see good reason why we should only allow -1 and 0 (and not only 0 (alternatively every non-positive number)).

Reply:
fair enough, I'll refactor these + the watch cache heuristic.